test/support/utils_invoke.h

```diff
@@ -219,6 +219,17 @@ struct invoke_on_all_hetero_policies
             auto my_policy = make_new_policy<kernel_name>(queue);
             iterator_invoker<::std::random_access_iterator_tag, /*IsReverse*/ ::std::false_type>()(
                 my_policy, op, ::std::forward<Args>(rest)...);
+
+#if !__SYCL_UNNAMED_LAMBDA__
+            volatile bool always_false = false;
+            if (always_false)
+            {
+                // We just need to compile some Kernel code and we don't need to run this code in run-time
+                // so we can move the rest of params again
+                iterator_invoker<::std::random_access_iterator_tag, /*IsReverse*/ ::std::false_type>()(
+                    std::move(my_policy), op, ::std::forward<Args>(rest)...);
```

Comment:
Have you considered introduction of a dedicated test which calls ~3-5 algorithms and resembles a reproducer from #2041, or extending `lambda_naming.pass`?

What bothers me in that approach is that this change is hidden inside the testing framework and it affects the majority of tests.

Reply:
I think this change exactly inside the testing framework gives us big benefit: we will check a lot of tests on Kernel compile phase.
About your suggestion of extending `lambda_naming.pass` and etc. - this mean we should rewrite all tests, I think.
Looks overcomplicated.

Reply:
We doesn't need have additional tests I think because we just not execute them in run-time, only compile.

Reply:
I see your point: this approach covers a lot and costs almost nothing.
As far as I understand, it will result in extra code to compile, and with your other PR merged - only host code.

My point: this is a niche problem, it is enough to have a dedicated test, even a compile-time test.

Reply:
I think the goal of this is not only to test regression of a specific problem, but also protect against an easy to introduce type of bug: Using different flavors of undecayed ExecutionPolicy in kernel submitters / kernel names, resulting in clashing names and extra JIT compilation time.

This is to lower future maintenance, and hopefully to allow us to keep perfect forwarding of policies where possible in the library, since we have this bug preventative measure.  For me, I see this as an alternative to #2101.  Although, I believe @SergeyKopienko may disagree and want to go forward with both.